content/post/against-ab-tests.md

```diff
@@ -0,0 +1,93 @@
+
+In this post I will provide a motivating example and then talk through some of
+the challenges that will come with building systems that can adapt to this new
+reality. I will not discuss any of the underlying statistics associated with
+building these types of systems ([the stitchfix post](https://multithreaded.stitchfix.com/blog/2018/11/08/bandits/) and [this post from
+google](https://support.google.com/analytics/answer/2846882?hl=en#) are great places to start for that), but instead will discuss what I see as opportunities as a strategic and architectural level.  
+## Motivating Example
+
+Just to convince you that this is important, let's work through a short numeric example. While these numbers are fictional, they harmonize with examples I have seen countless times during the real-life evaluation of A/B tests.
+
+Yet Another Mattress Company (YAMC) sells mattresses online (you may have seen their advertisements on the subway). They want to test a new version of their checkout flow that is optimized for mobile. The designers are a little bit concerned that while this new version is less clunky, it also conveys less information during the checkout flow, so it could hurt conversion on desktop.
+
+The team launches the test, and here are the results:
+
+| Test Arm         | Impressions | Conversions | Conversion Rate |
+|------------------|-------------|-------------|-----------------|
+| Mobile Optimized | 10000       | 300         | 3.0%            |
+| Old Version      | 10000       | 300         | 3.0%            |
+
+Rats! No difference. On a hunch, you decide to split the results by mobile and desktop traffic.
+
+| Device Type | Test Arm         | Impressions | Conversions | Conversion Rate |
+|-------------|------------------|-------------|-------------|-----------------|
+| Mobile      | Mobile Optimized | 3000        | 250         | 8.3%            |
+| Mobile      | Old Version      | 3000        | 100         | 3.3%            |
+| Desktop     | Mobile Optimized | 2000        | 50          | 2.5%            |
+| Desktop     | Old Version      | 2000        | 200         | 10%             |
+
+Wow! The new version ... did exactly what the designers expected! It made things better for mobile users and worse for desktop users[^1].
+
+It is really too bad that our A/B test showed no effect. I guess we'll have to send our designers back to the drawing board...
+
+But wait! What if we serve the mobile-optimized version to customers browsing on mobile? And serve the desktop-optimized version to customers browsing on desktop? What if we found a landing page that worked better on weekends when people have more time read? What if we find advertising copy that works well in California but not in Massachusetts? 
```

Comment:
When do you stop splitting the population? When do you stop dedicating resources to more versions? When the numbers get too small? 

As you know, when the numbers are big enough you can see a significant effect in any test. But is it meaningful?

Reply:
Yeah, i'm definitely eliding a lot of complexity here. I talk about this a bit at the bottom when I talk about the need for more statistical tools to answer these questions.

AFAICT this is the hardest part about implementing contextual bandits effectively -- how do you measure and propagate your uncertainty correctly?